### PR TITLE
add snakemake chapter

### PR DESCRIPTION
**Pull request by jhidding**
This removes the old chapter 5, which was useless anyway. Instead we get a Snakemake chapter. The idea is that anyone who teaches this should prepare an example workflow for themselves; one they're comfortable presenting.

There is one generic exercise in here, but I'm not sure if it runs on Windows.

### CODE REVIEW

File: _episodes/05-snakemake.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+- "Understand the operating logic (or semantics) of Snakemake."
+- "Create a parallel workflow from an existing script."
+keypoints:
+- "Snakemake is ideal for specifying workflows with large granularity."
+- "Dependencies are automatically resolved; Snakemake is 'demand driven' programming."
+- "You may freely intermix shell commands with Python code."
+- "Results are persistent: you can resume computations when needed."
+---
+
+> ## Metalesson!
```
Comment (PabRod) [inline _episodes/05-snakemake.md:19]: I don't think having _metalessons_ goes along very well with the philosophy of The Carpentries. Worst-case scenario, we can tag this as an incomplete episode, and keep working on it (i.e.: this is not a reason for rejecting this pull request).

File: _episodes/05-snakemake.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+> This is a metalesson! The instructor should fill in the details.
+>
+> The material in this demo is currently geared towards an astrophysics application. It is not essential that you understand the specifics of this application. That being said: it is much encouraged to use an example that the instructor is familiar with. The choice of a "real world" example is deliberate: we want participants to feel that this is a real application with real benifits. If you use a different demo, **please consider contributing** it to our repository. Ideally, future instructors can then choose a prepared demo that best suits their taste or background.
+{: .callout}
+
+# Setup
+> ## Preparation
+> To follow this tutorial, participants need to have cloned the repository at [github.com/escience-academy/parallel-python-workshop](https://github.com/escience-academy/parallel-python-workshop/). This repository includes an `environment.yml` file for creating the proper Conda environment, as well as the Jupyter notebook and `Snakefile` used in this demo.
+> This environment includes the GraphViz and ImageMagick packages needed to do some of the visualizations in this tutorial.
```
Comment (PabRod) [inline _episodes/05-snakemake.md:28]: Could be nice to add a link to Graphviz (https://graphviz.org/) and ImageMagick (https://imagemagick.org/).

File: _episodes/05-snakemake.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+### Run using multiple workers
+~~~bash
+time snakemake -j4
+~~~
+
+This should run significantly faster than the single worker case.
+
+# Specifics
+## Cosmic structure
+This workflow is located in the `cosmic_structure` folder in the `parallel-python-workshop` repository.
```
Comment (PabRod) [inline _episodes/05-snakemake.md:139]: The `cosmic_structure` folder currently doesn't exist (or at least I cannot find it). Would you please add it and link it in the text?

**Review by PabRod — APPROVED** [review]
Hi there,

the only comment that requires urgent action is the one about the absence of the `cosmic_structure` folder.

File: _episodes/05-snakemake.md (hunk 2 above)
Comment (PabRod) [reply _episodes/05-snakemake.md]: Solved in commit https://github.com/carpentries-incubator/lesson-parallel-python/pull/45/commits/b2a0585c86faeaa825db49b6ba7f19e87cd78450

File: _episodes/05-snakemake.md (hunk 1 above)
Comment (jhidding) [reply _episodes/05-snakemake.md:19]: I agree, this is just a push for anyone who doesn't agree to contribute more examples. An instructor can then take their pick. As a counter argument though: anyone teaching this should be experienced enough to be able to setup an example themselves.

File: _episodes/05-snakemake.md (hunk 3 above)
Comment (jhidding) [reply _episodes/05-snakemake.md:139]: I have this folder visible as "added 21 hours ago" ;)

Comment (PabRod) [reply _episodes/05-snakemake.md]: Solved https://github.com/carpentries-incubator/lesson-parallel-python/pull/45/commits/3e2dbdeacc6f15d87eb3ebd118057113a6cd723a

**Pull request merged**